invenio_rdm_records/resources/serializers/iiif/schema.py

```diff
@@ -209,3 +214,17 @@ def sortcanvases(self, manifest, many, **kwargs):
         """
         # manifest["sequences"][0]["canvases"].sort(key=lambda x: x["@id"])
         return manifest
+
+    @pre_dump
+    def add_iiif_links(self, data, many):
+        """Add IIIF links to each files entry."""
+        if "entries" in data["files"]:
+            for file in data["files"]["entries"].values():
+                if "links" not in file:
+                    links_tpl = current_rdm_records_service.files.file_links_item_tpl(
+                        data["id"]
+                    )
+                    obj = SimpleNamespace(**file)
+                    links = links_tpl.expand(system_identity, obj)
```

Comment:
Question: What is the impact of using the ``system_identity`` here? How does this behave with restricted files - i.e. is it potentially leaking information about restricted files to unauthorised users?